linux2mqtt/metrics.py

```diff
@@ -581,6 +581,96 @@ def poll(self, result_queue: Queue[BaseMetric]) -> bool:
         return True  # Expect a deferred result
 
 
+class NetConnectionMetrics(BaseMetric):
+    """Network connections metric."""
+
+    _name = "Network Connections"
+    icon = "mdi:ip-network"
+    device_class = ""
+    unit_of_measurement = "connections"
+    state_field = "count"
+
+    def __init__(self) -> None:
+        """Extract local IPs for evaluation during poll."""
+        super().__init__()
+        interface_addrs = psutil.net_if_addrs()
+        self.ips = set()
+        self.interval = 10
+
+        for snicaddrs in interface_addrs.values():
+            for snicaddr in snicaddrs:
+                if snicaddr.family.value in (2, 10):
+                    self.ips.add(snicaddr.address)
+
+    def poll(self, result_queue: Queue[Self]) -> bool:
+        """Poll new data for the network connection metrics.
+
+        Parameters
+        ----------
+        result_queue
+            (Unused)
+
+        Returns
+        -------
+        bool
+            True as the data is readily available
+
+        Raises
+        ------
+        Linux2MqttMetricsException
+            network connection information could not be gathered or prepared for publishing
+
+        """
+        try:
+            st = psutil.net_connections()
+            listening_ports = set([
+                x.laddr.port
+                for x in st
+                if x.status == "LISTEN" and x.laddr.ip in ("0.0.0.0", "::")
+            ])
```

Comment:
```suggestion
            listening_ports = {
                x.laddr.port
                for x in st
                if x.status == "LISTEN" and x.laddr and x.laddr.ip in ("0.0.0.0", "::")
            }
```

This should take care of the typing errors for example